\documentclass[12pt,t,aspectratio=169]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Identifying sample mix-ups in eQTL data}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{https://kbroman.org/Talk_OSGA2021}{\tt \scriptsize
  \color{foreground} kbroman.org/Talk\_OSGA2021}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-3mm}

  \note{These are slides for a talk for the OSGA seminar series on 11
    June 2021.

    Source: {\tt https://github.com/kbroman/Talk\_OSGA2021} \\
    Slides: {\tt https://kbroman.org/Talk\_OSGA2021}
}
} }


\begin{frame}[c]{Associations in systems genetics}
  \figh{Figs/triple_asso.pdf}{0.95}
\end{frame}


\begin{frame}[c]{Sample mix-ups}

  \vspace{2pt}

  \figh{Figs/b6btbr_plates.pdf}{0.90}

  \vspace{-12pt}
  \hfill \href{https://doi.org/10.1534/g3.115.019778}{\lolit
    \scriptsize Broman et al. (2015) {\tt doi:10.1534/g3.115.019778}}

\end{frame}


\begin{frame}[c]{More sample mix-ups}

  \vspace{2pt}

  \figh{Figs/b6btbr_expr_swaps.pdf}{0.85}

  \hfill \href{https://doi.org/10.1534/g3.115.019778}{\lolit
    \scriptsize Broman et al. (2015) {\tt doi:10.1534/g3.115.019778}}

\end{frame}



\begin{frame}[c]{Westra et al. (2011)}

\vspace{10mm}

\figw{Figs/westra_table2.png}{1.0}

  \vspace{10mm}
  \hfill \href{https://doi.org/10.1093/bioinformatics/btr323}{\lolit
    \scriptsize Westra et al. (2011) {\tt doi:10.1093/bioinformatics/btr323}}

\end{frame}


\begin{frame}[c]{Outline}

  \bbi
\item Sample duplicates
\item Sex verification
\item mRNA $\leftrightarrow$ protein
\item mRNA $\leftrightarrow$ DNA
\item protein $\leftrightarrow$ DNA
  \ei

\end{frame}




\begin{frame}[c]{}

  \centering
  \LARGE

  {\color{title} but first,} \\[24pt]

  {\vhilit Missing Data}

\end{frame}










\begin{frame}[c]{References}

  \bbi

  \item Westra et al. (2011) MixupMapper: correcting sample mix-ups in
    genome-wide datasets increases power to detect small genetic
    effects. Bioinformatics 15:2104--2111
    \href{https://doi.org/10.1093/bioinformatics/btr323}{\tt doi:10.1093/bioinformatics/btr323}

  \item Lynch et al (2012) Calling sample mix-ups in cancer population
    studies. PLOS One 7:e41815
    \href{https://doi.org/10.1371/journal.pone.0041815}{\tt doi:10.1371/journal.pone.0041815}

  \item Broman et al. (2015) Identification and correction of sample
    mix-ups in expression genetic data: A case study. G3 (Bethesda)
    5:2177--2186
    \href{https://doi.org/10.1534/g3.115.019778}{\tt doi:10.1534/g3.115.019778}

  \item Broman et al. (2019) Cleaning genotype data from Diversity
    Outbred mice. G3 (Bethesda) 9:1571--1579
    \href{https://doi.org/10.1534/g3.119.400165}{\tt doi:10.1534/g3.119.400165}

  \ei

\end{frame}


\begin{frame}[c]{}

\Large

Slides: \href{https://kbroman.org/Talk_OSGA2021}{\tt
  kbroman.org/Talk\_OSGA2021} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{7mm}

\href{https://kbroman.org}{\tt \lolit kbroman.org}

\vspace{7mm}

\href{https://github.com/kbroman}{\tt \lolit github.com/kbroman}

\vspace{7mm}

\href{https://twitter.com/kwbroman}{\tt \lolit @kwbroman}


\end{frame}

\end{document}
