\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Identifying sample mix-ups in eQTL data}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{https://kbroman.org/Talk_OSGA2021}{\tt \scriptsize
  \color{foreground} kbroman.org/Talk\_OSGA2021}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-3mm}

  \note{These are slides for a talk for the OSGA seminar series on 11
    June 2021.

    Source: {\tt https://github.com/kbroman/Talk\_OSGA2021} \\
    Slides: {\tt https://kbroman.org/Talk\_OSGA2021/osga2021.pdf} \\
    Slides with notes: {\tt https://kbroman.org/Talk\_OSGA2021/osga2021\_notes.pdf}

}
} }


\begin{frame}[c]{Associations in systems genetics}
  \figh{Figs/triple_asso.pdf}{0.95}

  \note{
    Systems genetics is all about associations between different
    datasets. It's critical, then, that the sample labels are correct
    for all data sets. As projects become larger and involve more
    groups of scientists, there's a greater chance for the
    introduction of errors in the sample labels.

    Sample duplicates, mixtures, and mix-ups will all weaken
    associations and so reduce the quality of the study results.

    On the other hand, with high-throughput genomic phenotypes, there
    is often the opportunity to both identify sample mix-ups and
    correct them.
  }

\end{frame}


\begin{frame}[c]{Sample mix-ups}

  \vspace{2pt}

  \figh{Figs/b6btbr_plates.pdf}{0.90}

  \vspace{-12pt}
  \hfill \href{https://doi.org/10.1534/g3.115.019778}{\lolit
    \scriptsize Broman et al. (2015) {\tt doi:10.1534/g3.115.019778}}

  \note{
    Here's an example of a set of mix-ups in the DNA samples for a
    project. In a mouse intercross with about 500 samples, there were
    nearly 20\% mix-ups. The dots indicate that the correct sample was
    in the correct place. The arrows point from where a sample should
    have been to where it was actually found.

    In this project, we had gene expression microarray data from six
    different tissues; that allowed us to identify and correct these
    errors.
  }

\end{frame}


\begin{frame}[c]{More sample mix-ups}

  \vspace{2pt}

  \figh{Figs/b6btbr_expr_swaps.pdf}{0.85}

  \hfill \href{https://doi.org/10.1534/g3.115.019778}{\lolit
    \scriptsize Broman et al. (2015) {\tt doi:10.1534/g3.115.019778}}

  \note{
    The mRNA samples had mix-ups, too. There were errors in each of
    the six tissues.
  }

\end{frame}



\begin{frame}[c]{Westra et al. (2011)}

\vspace{10mm}

\figw{Figs/westra_table2.png}{1.0}

  \vspace{10mm}
  \hfill \href{https://doi.org/10.1093/bioinformatics/btr323}{\lolit
    \scriptsize Westra et al. (2011) {\tt doi:10.1093/bioinformatics/btr323}}

  \note{
    Westra et al. (2011) was among the first to identify this
    potential problem and suggest a formal solution. They applied
    their approach to a number of public data sets and identified
    problems in most of them, including a study with 20\% mix-ups.
  }

\end{frame}


\begin{frame}[c]{Outline}

  \bbi
\item Sample duplicates
\item Sex verification
\item Sample mix-ups:
\bi
\item mRNA $\leftrightarrow$ protein
\item mRNA $\leftrightarrow$ DNA
\item protein $\leftrightarrow$ DNA
\ei
  \ei

  \note{
    If you have high-throughput, low-level phenotypes, you should at
    least attempt to identify potential sample mix-ups. My goal in this talk is to
    make it clear how to do this, to help ensure that this becomes a
    routine part of the data cleaning procedures in eQTL analyses.
  }

\end{frame}



{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

  \centering
  \LARGE
  \color{revforeground}

  But first \\[24pt]

  {\vhilit Missing Data}

  \note{
    Before you do anything, you should look at the amount of missing
    data, as this is often an important indication of sample quality.
  }

\end{frame}
}

\begin{frame}[c]{Percent missing genotypes}

\figh{Figs/missing_data.pdf}{0.95}

  \note{
    Here's a diversity outbred mouse project with 500 mice. Five
    samples had $>$ 25\% missing data and almost surely need to be
    omitted. A couple of samples have around 10\% missing data and
    might be recoverable but are still worth watching.

    Note that I'll be using a variety of data in this talk, but I
    won't be explaining where it's from. But I thank my collaborators
    for the data.
  }

\end{frame}



\begin{frame}[c]{Heterogeneous Stock/Diversity Outbreds}

\figh{Figs/hs_light.pdf}{0.87}

\note{
  I'm considering two datasets here. I won't say anything about
  where they're from, but they're both on diversity outbred mice, one
  with about 500 individuals and the other with 800 (though only about
  500 with gene expression data and 200 with proteomics data).

  You maybe can tell the sources, but let's not talk about the
  particular datasets right now. I'm working on a tutorial on
  identifying sample mix-ups but have agreed to wait to post it; I'll
  post it once an article correction appears.
}

\end{frame}



{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}


\centering
\LARGE
\color{revforeground}

Sample duplicates

\note{
  The next thing to look for is sample duplicates.
  Are there pairs of individuals with too-similar genotypes?

  These are pretty common. I don't know anything about monozygotic
  twins among mice, but we've always assumed that these are cases of
  sample duplication or contamination.
}

\end{frame}
}


\begin{frame}[c]{Percent matching genotypes}

\figh{Figs/hist_compare_geno.pdf}{0.95}

\note{
  It's simple to look for duplicate DNA samples: just calculate the
  proportion of matching genotypes for each pair of samples, and look
  for pairs that have very similar data.

  Here, we see no close matches. There's a group with rather low
  sharing, which are due to a couple of bad DNA samples, plus a group
  with somewhat above-normal sharing, which are likely siblings (these
  are again diversity outbred mice).

  This technique only works well for organisms with a lot of
  chromosomes. It would be hard to do this in Drosophila, because the
  variation in the ``just by chance'' sharing would be really high and
  cover the full rather 0--100\%.
}

\end{frame}





\begin{frame}[c]{Correlation between mRNA samples}

\figh{Figs/mrna_dups.png}{0.85}

\note{
  It seems like you should be able to do the same thing with mRNA or
  protein samples: just look at the correlation between samples, or
  perhaps the RMS difference. But I've not had much success finding
  duplicate samples this way. You maybe need to exclude genes that
  appear to not be expressed (and so are just noise).

  These are histograms of the correlation between mRNA expression
  samples. The two sexes are anti-correlated. The lower histogram is
  for correlation between measurements after controlling for sex.
}

\end{frame}







\begin{frame}[c]{Correlation between protein samples}

\figh{Figs/protein_dups.png}{0.85}

\note{
  This is similar as the last slide, but with mass-spec-based protein
  measurements. The anti-correlation between sexes is not as strong
  but still present.
}

\end{frame}







{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}


\centering
\LARGE
\color{revforeground}

Sex verification

\note{
  One way to identify sample mislabelings is by comparing the
  annotated sex to what you can infer from the genotypes or
  expression data.
}

\end{frame}
}



\begin{frame}[c]{X and Y genotype dosage}

\figh{Figs/xydosage.pdf}{0.95}

\note{
   Historically, I would look at heterozygosity on the X chromosome
   to verify sex. But even better, for verifying sex in the genotype
   data, is to look at the dosage of X and Y chromosome markers
   (average intensity for microarray-based genotypes, or frequency of
   mapped reads for sequencing-based genotypes).

   The x-axis is average intensity of SNPs on the X chromosome; the
   y-axis is average intensity of SNPs on the Y chromosome.

   The green ball in the lower-right are females with two X
   chromosomes and no Y. The purple ball in the upper-left are males
   with one X and one Y. The points in the lower-left are maybe XO
   females.

   We are looking for females in the upper-left (and there is one
   such) or males in the lower-right.
}

\end{frame}



\begin{frame}[c]{Sex and gene expression}


\figh{Figs/sex_mrna_indgenes.pdf}{0.95}


  \note{
    We should be able to do the same sort of thing, to verify the sex
    of the mRNA samples.

    We could think hard about genes that should show a big sex
    difference (for example, \emph{Xist}), or we could just do t-tests
    to identify a set of genes that have big sex differences. These
    are the top six genes, in terms of sex difference in expression.
    Of course, Xist is the first.

    We could choose the top say 100 genes and use them to form a
    classifier for sex from gene expression.
    We want a method that can handle some misclassified samples.
  }
\end{frame}




\begin{frame}[c]{Sex and gene expression}


\figh{Figs/sex_mrna.pdf}{0.95}


  \note{
    What I ended up doing was just pick the top 100 genes and then
    calculate the mean in males and females, and then for each sample,
    look at the RMS difference from the male means and from the female
    means. This is a plot of those.

    Most of the samples have well-differentiated sex by this approach,
    but there are a bunch of samples in the middle, maybe due to batch
    differences that I've not accounted for?

    Anyway, while there are a number of samples in the middle that are
    unclear, there's also a set of males that look like females and
    one female that looks like male.
  }
\end{frame}





\begin{frame}[c]{Sex and proteins}


\figh{Figs/sex_protein_indgenes.pdf}{0.95}


  \note{
    Protein levels also show big sex differences, so we can do the
    same thing again with proteins.
  }
\end{frame}




\begin{frame}[c]{Sex and proteins}


\figh{Figs/sex_protein.pdf}{0.95}


  \note{
    I did the same thing as with the mRNA data: pick the top 100
    proteins by their sex difference, calculate the average for males
    and for females, and then take the RMS difference for each sample
    from the male means and the female means.

    For this particular data set, there are a few females
    deep within the males.
  }
\end{frame}








{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

\centering
\LARGE
\color{revforeground}

Sample mix-ups

\bigskip

mRNA $\leftrightarrow$ protein

\note{
  We now turn to the question of sample mix-ups more directly. We'll
  start by comparing a set of mRNA expression data to a set of
  proteomics data.
}

\end{frame}
}


\begin{frame}[c]{mRNA $\leftrightarrow$ protein method}

  \only<1|handout 0>{\figh{Figs/evp_mixup_scheme1.pdf}{0.95}}
  \only<2|handout 0>{\figh{Figs/evp_mixup_scheme2.pdf}{0.95}}
  \only<3|handout 0>{\figh{Figs/evp_mixup_scheme3.pdf}{0.95}}
  \only<4>{\figh{Figs/evp_mixup_scheme.pdf}{0.95}}


  \note{
    We have two rectangles of data where some of the rows are supposed
    to correspond.

    We first look for gene/protein pairs that are highly correlated.
    We then focus on those and calculate the correlation between rows,
    among the highly correlated gene/protein pairs.

    The between-sample correlations can be viewed as a similarity
    matrix. We're hoping to see a single large value in each row, for
    the samples that are supposed to correspond.
   }

\end{frame}


\begin{frame}[c]{mRNA $\leftrightarrow$ protein correlations}

  \figh{Figs/hist_cor_mrna_prot.pdf}{0.95}

  \note{
    For a set of gene expression data and proteomics data, I
    considered each protein and found the most-correlated gene,
    adjusting for sex. (Whether to adjust for sex is not totally
    clear to me, but it makes the subsequent figures easier to
    understand.)

    I show the top protein/gene pair (which all correspond to an mRNA
    and its protein product).
  }
\end{frame}


\begin{frame}[c]{mRNA $\leftrightarrow$ protein similarity matrix}

  \figw{Figs/heatmap_mrna_v_protein.png}{1.0}

  \note{
    We take the top 100 gene/protein pairs and use them to calculate
    the correlation between samples. This is that correlation matrix.
    There are not quite 200 protein samples and a bit over 400 mRNA
    samples.

    Black means more similar, and white less similar. The checkboard
    pattern is probably some sort of batch effects. If I hand't
    controlled for sex, there would be two big squares here, for the
    two sexes.

    You can see a couple of diagonal lines, for the samples that
    correspond.
  }
\end{frame}



\begin{frame}[c]{mRNA $\leftrightarrow$ protein similarities}

  \figh{Figs/self_nonself_mrna_prot.pdf}{0.95}

  \note{
    If we separate the values for samples that are supposed to correspond and the
    values for samples that are not supposed to, we ideally would have
    two non-overlapping distributions.

    The bulk of the samples are like that: large values for
    the self-self similarities, and small values for the self-nonself
    similarities. But there are a bunch of self-self values that are
    small, and a few self-nonself values that are large.
  }
\end{frame}



\begin{frame}[c]{mRNA $\leftrightarrow$ protein: closest vs self}

  \figh{Figs/self_v_best_mrna_prot.pdf}{0.95}

  \note{
    Here's a scatterplot where for each protein sample I find the
    maximum similarity in that row and plot it against the similarity
    for the mRNA sample that is supposed to correspond.

    For most samples, these two values are the same, and they are
    large. (That's the diagonal line in the top-right.)

    But there are a bunch of samples where the self-similarity is
    small, indicating that it seems to have the wrong label. Many of
    these are largely similar to some other sample, but a few are not
    similar to anything.
  }
\end{frame}






\begin{frame}[c]{mRNA $\leftrightarrow$ protein: selected samples}

  \figh{Figs/samples_mrna_prot.pdf}{0.95}

  \note{
    Here I have focused on a selected set of six protein samples,
    and just plot all of the values in that row of the similarity
    matrix. For each of these protein samples, I highlight the closest
    mRNA sample.

    The pair on the left looks like a sample swap: M348 is closest to
    M410 and vice versa. Similarly, the pair in the middle looks like
    a sample swap, with F371 being closest to M349 and vice versa. The
    two samples on the right look to be correctly labeled.

    Overall, there were about 9 sample swaps and then a few other
    protein samples that didn't seem to correspond to any mRNA sample.

    With these sample swaps, we can't tell, from these data alone,
    whether the problem is with the mRNA samples, the protein samples,
    or some of each.
  }
\end{frame}









{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

\centering
\LARGE
\color{revforeground}

Sample mix-ups

\bigskip

DNA $\leftrightarrow$ mRNA

\note{
  Let's now turn to the same sort of thing, but comparing DNA-based
  genotypes to mRNA expression.
}

\end{frame}
}



\begin{frame}[c]{DNA $\leftrightarrow$ mRNA method}

  \only<1|handout 0>{\figh{Figs/gve_mixup_scheme1.pdf}{0.95}}
  \only<2|handout 0>{\figh{Figs/gve_mixup_scheme2.pdf}{0.95}}
  \only<3|handout 0>{\figh{Figs/gve_mixup_scheme3.pdf}{0.95}}
  \only<4>{\figh{Figs/gve_mixup_scheme.pdf}{0.95}}


  \note{
    The approach we take starts similarly: we identify a set of
    expression traits with strong eQTL, and then pull out the
    genotypes at those eQTL.

    The simplest thing is to use the genotype/expression
    correspondence to get predected expression values for each trait,
    based on the observed eQTL genotypes. Basically just calculate the
    average trait value for each genotype and assign that as the
    fitted value.

    We can then take the root-mean-square (RMS) difference between the
    predicted expression values for a DNA sample and the observed
    expression values for an mRNA sample, and used that as a distance
    matrix, comparing the DNA and mRNA samples.
  }

\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ mRNA LOD scores}

  \figh{Figs/hist_lod_mrna.pdf}{0.95}

  \note{
    For a set of RNA-seq based expression data, here are the LOD
    scores. For each expression trait, I did a genome scan adjusting
    for the sex and picked out the single genome-wide maximum value.
    There are 100 genes with LOD about 150 and higher.
  }
\end{frame}


\begin{frame}[c]{DNA $\leftrightarrow$ mRNA distance matrix}

  \figw{Figs/heatmap_dna_v_mrna.png}{1.0}

  \note{
    Using those 100 genes, I calculated predicted expression values
    for each DNA sample, and then calculated this distance matrix.
    Black again means close. You can see a few black diagonal lines in
    here. There are about 450 mRNA samples and a bit over 800 DNA
    samples.
  }
\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ mRNA distances}

  \figh{Figs/self_nonself_dna_mrna.pdf}{0.95}

  \note{
    Here, if I pull apart the self-self and self-nonself distances, I
    find that there's almost no overlap.
  }
\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ mRNA: closest vs self}

  \figh{Figs/self_v_best_dna_mrna.pdf}{0.95}

  \note{
    If I calculate the minimum distance in each row and plot it
    against the self-self distance, I found that each mRNA sample is
    actually closest to itself than to any other sample. There don't
    seem to be any problems here.
  }
\end{frame}






\begin{frame}[c]{DNA $\leftrightarrow$ mRNA: selected samples}

  \figh{Figs/samples_dna_mrna.pdf}{0.95}

  \note{
    Here are those six selected samples again. In each case, the
    samples look to have the correct labels.
  }
\end{frame}















{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

\centering
\LARGE
\color{revforeground}

Sample mix-ups

\bigskip

DNA $\leftrightarrow$ protein

\note{
  Let's turn to a comparison between DNA-based genotypes and
  proteomics data.
}

\end{frame}
}


\begin{frame}[c]{DNA $\leftrightarrow$ protein method}

  \figh{Figs/gvp_mixup_scheme.pdf}{0.95}


  \note{
    The approach is the same as for the comparison of DNA and mRNA.
    For each protein, I do a genome scan and look for the strongest
    pQTL, and I pull out the proteins that have strong pQTL.

    I then use the association between genotype and protein level to
    get predicted protein levels for each DNA sample. I then calculate
    the RMS difference between the predicted protein levels for a
    DNA sample and observed protein levels for some proteomics sample,
    and use this as a distance matrix.
   }

\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ protein correlations}

  \figh{Figs/hist_lod_prot.pdf}{0.95}

  \note{
    There are many proteins with very strong pQTL. I'll focus on the
    top 100 proteins, with LOD scores about 30 and up.
  }
\end{frame}


\begin{frame}[c]{DNA $\leftrightarrow$ protein distance matrix}

  \figw{Figs/heatmap_dna_v_prot.png}{1.0}

  \note{
    Here's the distance matrix comparing just under 200 proteomics
    samples to just over 800 DNA samples. Black means close; you can
    see two prominent diagonal lines.
  }
\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ protein distances}

  \figh{Figs/self_nonself_dna_prot.pdf}{0.95}

  \note{
    If we separate the self-self distances from the self-nonself ones,
    we again see a bunch of self-self distances that are overly large,
    and self-nonself distances that are too small.
  }
\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ protein: closest vs self}

  \figh{Figs/self_v_best_dna_prot.pdf}{0.95}

  \note{
    Taking the minimum distance in each row and plotting that against
    the distance for the sample with the same label, we see that most
    samples look to be correctly labeled, but there's a big group that
    are not close to the corresponding sample but are close to some
    other sample.
  }
\end{frame}






\begin{frame}[c]{DNA $\leftrightarrow$ protein: selected samples}

  \figh{Figs/samples_dna_prot.pdf}{0.95}

  \note{
    Here are those six selected samples again. A couple of sample
    swaps and a pair that are correct.

    Combining what we've learned from the DNA/mRNA/protein, we can
    conclude that the mislabeling problems were in the protein
    samples.

    Overall, we identified 9 sample swaps and two additional
    mislabeled samples. There were 192 total samples, so this is just
    above 10\% problems.
  }
\end{frame}







\begin{frame}[c]{Summary}

  \begin{columns}[T]

    \begin{column}[T]{0.5\textwidth}

      \bbi
    \item This shouldn't happen.
    \item But if it does, you should find it.
    \item If two data sets have rows that correspond, you should
      check that they {\vhilit do} correspond.
      \ei
    \end{column}


    \begin{column}[T]{0.5\textwidth}

      \vspace*{-0.1\textheight}
      \hspace*{0.273\textwidth} \figw{Figs/b6btbr_plates.pdf}{1.8}

    \end{column}

    \end{columns}

  \note{
    In summary: sample mix-ups shouldn't happen; we should do our best
    to avoid them. But if the \emph{do\/} happen, we should find them.

    It's not too hard to look for sample mix-ups. Pull out pairs of
    highly correlated variables and then use them to establish the
    similarity or distance between samples.
  }
\end{frame}



\begin{frame}[c]{References}

  \bbi

  \item Westra et al. (2011) MixupMapper: correcting sample mix-ups in
    genome-wide datasets increases power to detect small genetic
    effects. Bioinformatics 15:2104--2111
    \href{https://doi.org/10.1093/bioinformatics/btr323}{\tt doi:10.1093/bioinformatics/btr323}

  \item Lynch et al (2012) Calling sample mix-ups in cancer population
    studies. PLOS One 7:e41815
    \href{https://doi.org/10.1371/journal.pone.0041815}{\tt doi:10.1371/journal.pone.0041815}

  \item Broman et al. (2015) Identification and correction of sample
    mix-ups in expression genetic data: A case study. G3 (Bethesda)
    5:2177--2186
    \href{https://doi.org/10.1534/g3.115.019778}{\tt doi:10.1534/g3.115.019778}

  \item Broman et al. (2019) Cleaning genotype data from Diversity
    Outbred mice. G3 (Bethesda) 9:1571--1579
    \href{https://doi.org/10.1534/g3.119.400165}{\tt doi:10.1534/g3.119.400165}

  \ei

  \note{
    Here are some relevant references. The Lynch et al. (2012) paper has
    some useful comments about experimental design.
   }

\end{frame}


\begin{frame}[c]{}

\Large

Slides: \href{https://kbroman.org/Talk_OSGA2021}{\tt
  kbroman.org/Talk\_OSGA2021} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{7mm}

\href{https://kbroman.org}{\tt \lolit kbroman.org}

\vspace{7mm}

\href{https://github.com/kbroman}{\tt \lolit github.com/kbroman}

\vspace{7mm}

\href{https://twitter.com/kwbroman}{\tt \lolit @kwbroman}

  \note{
    Here is where you can find me and my slides.
   }

\end{frame}



\begin{frame}[c]{DNA $\leftrightarrow$ protein: best vs 2nd-best}

  \figh{Figs/best_v_2ndbest_dna_prot.pdf}{0.95}

  \note{
    Here's an extra slide showing (on the right) the best distance
    vs the 2nd-best distance. The panel on the left we'd seen before,
    of the best distance vs the minimum distance.

    This is useful for assessing the evidence that the sample with the
    best distance is the real sample identity. If the best distance is
    quite far from the second-best distance, than it seems like we can
    identify the true sample. If they're similar, we don't have much
    evidence to relabel the sample.

    I've highlighted a couple of points in orange, where the
    second-best distance is quite close to the best distance. But for
    both of these, the best distance is the same as the self distance,
    and so there's no reason to question their identity.
  }
\end{frame}


\end{document}
