\documentclass[12pt,t,aspectratio=169]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Identifying sample mix-ups in eQTL data}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{https://kbroman.org/Talk_OSGA2021}{\tt \scriptsize
  \color{foreground} kbroman.org/Talk\_OSGA2021}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-3mm}

  \note{These are slides for a talk for the OSGA seminar series on 11
    June 2021.

    Source: {\tt https://github.com/kbroman/Talk\_OSGA2021} \\
    Slides: {\tt https://kbroman.org/Talk\_OSGA2021/osga2021.pdf} \\
    Slides with notes: {\tt https://kbroman.org/Talk\_OSGA2021/osga2021\_notes.pdf}

}
} }


\begin{frame}[c]{Associations in systems genetics}
  \figh{Figs/triple_asso.pdf}{0.95}

  \note{
    Systems genetics is all about associations between different
    datasets. It's critical, then, that the sample labels are correct
    for all data sets. As projects become larger and involve more
    groups of scientists, there's a greater chance for the
    introduction of errors in the sample labels.

    Sample duplicates, mixtures, and mix-ups will all weaken
    associations and so reduce the quality of the study results.

    On the other hand, with high-throughput genomic phenotypes, there
    is often the opportunity to both identify sample mix-ups and
    correct them.
  }

\end{frame}


\begin{frame}[c]{Sample mix-ups}

  \vspace{2pt}

  \figh{Figs/b6btbr_plates.pdf}{0.90}

  \vspace{-12pt}
  \hfill \href{https://doi.org/10.1534/g3.115.019778}{\lolit
    \scriptsize Broman et al. (2015) {\tt doi:10.1534/g3.115.019778}}

  \note{
    Here's an example of a set of mix-ups in the DNA samples for a
    project. In a mouse intercross with about 500 samples, there were
    nearly 20\% mix-ups. The dots indicate that the correct sample was
    in the correct place. The arrows point from where a sample should
    have been to where it was actually found.

    In this project, we had gene expression microarray data from six
    different tissues; that allowed us to identify and correct these
    errors.
  }

\end{frame}


\begin{frame}[c]{More sample mix-ups}

  \vspace{2pt}

  \figh{Figs/b6btbr_expr_swaps.pdf}{0.85}

  \hfill \href{https://doi.org/10.1534/g3.115.019778}{\lolit
    \scriptsize Broman et al. (2015) {\tt doi:10.1534/g3.115.019778}}

  \note{
    The mRNA samples had mix-ups, too. There were errors in each of
    the six tissues.
  }

\end{frame}



\begin{frame}[c]{Westra et al. (2011)}

\vspace{10mm}

\figw{Figs/westra_table2.png}{1.0}

  \vspace{10mm}
  \hfill \href{https://doi.org/10.1093/bioinformatics/btr323}{\lolit
    \scriptsize Westra et al. (2011) {\tt doi:10.1093/bioinformatics/btr323}}

  \note{
    Westra et al. (2011) was among the first to identify this
    potential problem and suggest a formal solution. They applied
    their approach to a number of public data sets and identified
    problems in most of them, including a study with 20\% mix-ups.
  }

\end{frame}


\begin{frame}[c]{Outline}

  \bbi
\item Sample duplicates
\item Sex verification
\item mRNA $\leftrightarrow$ protein
\item mRNA $\leftrightarrow$ DNA
\item protein $\leftrightarrow$ DNA
  \ei

  \note{
    If you have high-throughput, low-level phenotypes, you should at
    least attempt to identify potential sample mix-ups. My goal in this talk is to
    make it clear how to do this, to help ensure that this becomes a
    routine part of the data cleaning procedures in eQTL analyses.
  }

\end{frame}



{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

  \centering
  \LARGE
  \color{revforeground}

  But first \\[24pt]

  {\vhilit Missing Data}

  \note{
    Before you do anything, you should look at the amount of missing
    data, as this is often an important indication of sample quality.
  }

\end{frame}
}

\begin{frame}[c]{Percent missing genotypes}

\figh{Figs/missing_data.pdf}{0.95}

  \note{
    Here's a diversity outbred mouse project with 500 mice. Five
    samples had $>$ 25\% missing data and almost surely need to be
    omitted. A couple of samples have around 10\% missing data and
    might be recoverable but are still worth watching.

    Note that I'll be using a variety of data in this talk, but I
    won't be explaining where it's from. But I thank my collaborators
    for the data.
  }

\end{frame}





{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}


\centering
\LARGE
\color{revforeground}

Sample duplicates

\note{
  The next thing to look for is sample duplicates.
  Are there pairs of individuals with too-similar genotypes?

  These are pretty common. I don't know anything about monozygotic
  twins among mice, but we've always assumed that these are cases of
  sample duplication or contamination.
}

\end{frame}
}


\begin{frame}[c]{Percent matching genotypes}

\figh{Figs/hist_compare_geno.pdf}{0.95}

\note{
  It's simple to look for duplicate DNA samples: just calculate the
  proportion of matching genotypes for each pair of samples, and look
  for pairs that have very similar data.

  Here, we see no close matches. There's a group with rather low
  sharing, which are due to a couple of bad DNA samples, plus a group
  with somewhat above-normal sharing, which are likely siblings (these
  are again diversity outbred mice).

  This technique only works well for organisms with a lot of
  chromosomes. It would be hard to do this in Drosophila, because the
  variation in the ``just by chance'' sharing would be really high and
  cover the full rather 0--100\%.
}

\end{frame}





\begin{frame}[c]{Correlation between mRNA samples}

\figh{Figs/mrna_dups.png}{0.85}

\note{
  It seems like you should be able to do the same thing with mRNA or
  protein samples: just look at the correlation between samples, or
  perhaps the RMS difference. But I've not had much success finding
  duplicate samples this way. You maybe need to exclude genes that
  appear to not be expressed (and so are just noise).

  These are histograms of the correlation between mRNA expression
  samples. The two sexes are anti-correlated. The lower histogram is
  for correlation between measurements after controlling for sex.
}

\end{frame}







\begin{frame}[c]{Correlation between protein samples}

\figh{Figs/protein_dups.png}{0.85}

\note{
  This is similar as the last slide, but with mass-spec-based protein
  measurements. The anti-correlation between sexes is not as strong
  but still present.
}

\end{frame}







{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}


\centering
\LARGE
\color{revforeground}

Sex verification

\note{
  One way to identify sample mislabelings is by comparing the
  annotated sex to what you can infer from the genotypes or
  expression data.
}

\end{frame}
}



\begin{frame}[c]{X and Y genotype dosage}

\figh{Figs/xydosage.pdf}{0.95}

\note{
   Historically, I would look at heterozygosity on the X chromosome
   to verify sex. But even better, for verifying sex in the genotype
   data, is to look at the dosage of X and Y chromosome markers
   (average intensity for microarray-based genotypes, or frequency of
   mapped reads for sequencing-based genotypes).

   The x-axis is average intensity of SNPs on the X chromosome; the
   y-axis is average intensity of SNPs on the Y chromosome.

   The green ball in the lower-right are females with two X
   chromosomes and no Y. The purple ball in the upper-left are males
   with one X and one Y. The points in the lower-left are maybe XO
   females.

   We are looking for females in the upper-left (and there is one
   such) or males in the lower-right.
}

\end{frame}



\begin{frame}[c]{Sex and gene expression}


\figh{Figs/sex_mrna.pdf}{0.95}


  \note{
    We should be able to do the same sort of thing, to verify the sex
    of the mRNA samples.

    We could think hard about genes that should show a big sex
    difference (for example, \emph{Xist}), or we could just do t-tests
    to identify a set of genes that have big sex differences, and then
    use them to form a classifier for sex from gene expression.
    We want a method that can handle some misclassified samples.

    We could also just calculate fitted values for each sex at each
    gene and then look at RMS of the residuals from the two.
  }
\end{frame}








{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

\centering
\LARGE
\color{revforeground}

mRNA $\leftrightarrow$ protein

\note{
}

\end{frame}
}






{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

\centering
\LARGE
\color{revforeground}

DNA $\leftrightarrow$ mRNA

\note{
}

\end{frame}
}







{\setbeamercolor{normal text}{bg=revbackground}

\begin{frame}[c]{}

\centering
\LARGE
\color{revforeground}

DNA $\leftrightarrow$ protein

\note{
}

\end{frame}
}













\begin{frame}[c]{References}

  \bbi

  \item Westra et al. (2011) MixupMapper: correcting sample mix-ups in
    genome-wide datasets increases power to detect small genetic
    effects. Bioinformatics 15:2104--2111
    \href{https://doi.org/10.1093/bioinformatics/btr323}{\tt doi:10.1093/bioinformatics/btr323}

  \item Lynch et al (2012) Calling sample mix-ups in cancer population
    studies. PLOS One 7:e41815
    \href{https://doi.org/10.1371/journal.pone.0041815}{\tt doi:10.1371/journal.pone.0041815}

  \item Broman et al. (2015) Identification and correction of sample
    mix-ups in expression genetic data: A case study. G3 (Bethesda)
    5:2177--2186
    \href{https://doi.org/10.1534/g3.115.019778}{\tt doi:10.1534/g3.115.019778}

  \item Broman et al. (2019) Cleaning genotype data from Diversity
    Outbred mice. G3 (Bethesda) 9:1571--1579
    \href{https://doi.org/10.1534/g3.119.400165}{\tt doi:10.1534/g3.119.400165}

  \ei

  \note{
    Here are some relevant references. The Lynch et al. (2012) paper has
    some useful comments about experimental design.
   }

\end{frame}


\begin{frame}[c]{}

\Large

Slides: \href{https://kbroman.org/Talk_OSGA2021}{\tt
  kbroman.org/Talk\_OSGA2021} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{7mm}

\href{https://kbroman.org}{\tt \lolit kbroman.org}

\vspace{7mm}

\href{https://github.com/kbroman}{\tt \lolit github.com/kbroman}

\vspace{7mm}

\href{https://twitter.com/kwbroman}{\tt \lolit @kwbroman}

  \note{
    Here is where you can find me and my slides.
   }

\end{frame}

p\end{document}
